File: 0001_TwoSum/TwoSum.cs

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> mem = new Dictionary<int, int>();
			foreach (var x in nums.Select((val, ind) => new {val, ind})) {
				if (mem.ContainsKey(x.val)) {
					return new int[]{mem[x.val], x.ind};
				}
				mem[target-x.val] = mem.ContainsKey(target-x.val) ? mem[target-x.val] : x.ind;
			}
			return new int[] {0,0};
    }
}

File: 0001_TwoSum/TwoSum.js
var twoSum = function (nums, target) {
    const dict = {}
    for (let i in nums) {
        if (dict[nums[i]]) {
            return [dict[nums[i]], i]
        } else {
            dict[target - nums[i]] = i;
        }
    }
};

twoSum([2, 7, 11, 15], 9)